Sources/ArcGISToolkit/Components/FeatureTemplatePicker/FeatureTemplatePicker.swift

```diff
@@ -0,0 +1,340 @@
+                if let image = info.image {
+                    Image(uiImage: image)
+                } else {
+                    Image(systemName: "minus")
+                        .foregroundStyle(.secondary)
+                }
+            }
+            Spacer()
+            if info == selection {
+                Image(systemName: "checkmark")
+            }
+        }
+        .contentShape(Rectangle())
+        .onTapGesture {
+            selection = info
+        }
+    }
+}
+
+extension FeatureTemplatePicker {
+    /// The model for the feature template picker.
+    @MainActor
+    final class Model: ObservableObject {
+        /// The associated geo model.
+        let geoModel: GeoModel
+        /// Include feature templates from tables where features cannot be created.
+        let includeNonCreatableFeatureTemplates: Bool
+        /// Search text for filtering the list of templates.
+        var searchText: String = "" {
+            didSet { templatesOrSearchTextDidChange() }
+        }
+        /// The complete unfiltered list of feature template sections.
+        private(set) var unfilteredFeatureTemplateSections = [FeatureTemplateSectionInfo]() {
+            didSet { templatesOrSearchTextDidChange() }
+        }
+        
+        /// The sections.
+        @Published private(set) var featureTemplateSections = [FeatureTemplateSectionInfo]()
+        /// A Boolean value indicating whether templates are being generated.
+        @Published private(set) var isGeneratingFeatureTemplates = false
+        /// A Boolean value indicating if content is unavailable.
+        @Published private(set) var showContentUnavailable = false
+        /// A Boolean value indicating if no templates were found with the given search text.
+        @Published private(set) var showNoTemplatesFound = false
+        
+        /// Creates a model for a given geo model.
+        init(geoModel: GeoModel, includeNonCreatableFeatureTemplates: Bool) {
+            self.geoModel = geoModel
+            self.includeNonCreatableFeatureTemplates = includeNonCreatableFeatureTemplates
+        }
+        
+        /// Generates the feature templates.
+        func generateFeatureTemplates() async {
+            isGeneratingFeatureTemplates = true
+            defer { isGeneratingFeatureTemplates = false }
+            
+            unfilteredFeatureTemplateSections = await makeFeatureTemplateSections()
+        }
+        
+        /// The feature template section information for the geo model.
+        private func makeFeatureTemplateSections() async -> [FeatureTemplateSectionInfo] {
+            try? await geoModel.load()
+            var sections = [FeatureTemplateSectionInfo]()
+            let layersAndTables = geoModel.arcGISFeatureLayersAndTables
+            await layersAndTables.map(\.table).load()
+            for (layer, table) in layersAndTables {
+                guard includeNonCreatableFeatureTemplates || table.canAddFeature else { continue }
```

Comment:
This fixes the crash I noted with unloaded tables.
```suggestion
                guard includeNonCreatableFeatureTemplates || table.canAddFeature,
                      table.loadStatus == .loaded,
                      layer.loadStatus == .loaded else { continue }
```

Reply:
Does the layer need to be loaded as well to fix this crash or only the table?

Reply:
Only the table for this specific crash. I added the layer for completeness, but not sure it's necessary.

Reply:
thanks